core/src/main/java/org/elasticsearch/index/translog/BufferingTranslogWriter.java

```diff
@@ -107,15 +119,24 @@ public void sync() throws IOException {
             return;
         }
         synchronized (this) {
+            ensureOpen();
```

Comment:
unless under lock, I don't think this buys us much(and we check in the flush method)?. we increment the channel reference here, so that might be enough as well.